WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift

```diff
@@ -558,6 +557,13 @@ extension WooAnalyticsEvent {
             return false
         }
 
+        private static func productTypes(order: Order, products: [Product]) -> String {
+            let productIDs = order.items.map { $0.productID }
+            return productIDs.compactMap { productID in
+                products.first(where: { $0.productID == productID })?.productType.rawValue
+            }.uniqued().sorted().joined(separator: ",")
```

Comment:
I think this products are already the products of the order, right?
if so, I think we can do only
```swift
products.map { $0.productType }.uniqued().sorted().joined(separator: ",")
```

Reply:
true for order creation, but for the `order_products_loaded` event in order details it's getting all the products from the site (after loading missing products remotely):

https://github.com/woocommerce/woocommerce-ios/blob/81cfa23cf81f9729d073fb35188682e2a60a89a0/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsResultsControllers.swift#L27

thus the filter is still needed to support these 2 events.